src/apme_engine/daemon/engine_server.py

```
@@ -3458,34 +3521,97 @@ async def Health(
                 channel = grpc.aio.insecure_channel(addr)
                 try:
                     stub = validate_pb2_grpc.ValidatorStub(channel)  # type: ignore[no-untyped-call]
                     resp = await stub.Health(HealthRequest(), timeout=5)
-                    downstream.append(ServiceHealth(name=name, status=resp.status, address=addr))
+                    status = resp.status
+                    if name in REQUIRED_VALIDATORS and status != "ok":
+                        unhealthy = True
+                    downstream.append(ServiceHealth(name=name, status=status, address=addr))
                 finally:
                     await channel.close(grace=None)
             except Exception as e:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)